### ISSUE TITLE
Meeseeks.html/1 doesn't correctly encode attributes with both single and double quotes

### ISSUE THREAD

**Issue by ericlathrop** [mischov/meeseeks#62]
Example:
```elixir
iex(19)> "<span data-stuff=\"&quot;'\">" |> Meeseeks.parse |> Meeseeks.html      
"<html><head></head><body><span data-stuff='\"''></span></body></html>"
```
Notice the extra single quote.

Expected either:
```elixir
"<html><head></head><body><span data-stuff='\"&#39;'></span></body></html>"
```
or
```elixir
"<html><head></head><body><span data-stuff=\"&quot;'\"></span></body></html>"
```

**Comment by ericlathrop** [comment]
[Floki actually has this same bug](https://github.com/philss/floki/issues/209), which is why I decided to try meeseeks.

**Comment by mischov** [comment]
Thank you for the report. I'll check it and the PR over and get back to you if changes are required.

**Comment by mischov** [comment]
@ericlathrop Verified the bug, and discovered an additional one since text nodes need to be encoded with some kind of escaping to prevent `<` (and probably `&`) so that

```elixir
"<span>&lt;script&gt;Hello&lt;/script&gt;</span>" 
```

doesn't round trip out to

```elixir
"<span><script>Hello</script></span>"
```

I am currently reviewing your solution and alternatives- I don't really like that something such as `<span data-stuff='double-quote"'>` which would have round-tripped identically before now comes back like `<span data-stuff=\"double-quote&quot;\"></span>`.

There is an argument to be made that such an approach would be the safe one, but I want a better idea of my possible alternatives before I make what amounts to a [breaking change](https://github.com/philss/floki/pull/183/files).

**Comment by mischov** [comment]
Going to go with a slimmed down version of what you proposed.

When encoding to html with `Meeseeks.html/1`:

- Attribute values will always be double quoted, and `&` and `"` will be escaped.
- Text will have `<`, `>`, and `&` escaped.

Thanks again for reporting this @ericlathrop, the lack of encoding was a serious problem.

**Comment by mischov** [comment]
Fixed in v0.12.0.
